mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -862,18 +866,9 @@ impl TakerSwap {
         let maker_coin_swap_contract_address = self.maker_coin.swap_contract_address();
         let taker_coin_swap_contract_address = self.taker_coin.swap_contract_address();
 
-        let maker_coin_htlc_key_pair = self.maker_coin.get_htlc_key_pair();
-        let taker_coin_htlc_key_pair = self.taker_coin.get_htlc_key_pair();
-
-        let maker_coin_htlc_pubkey = match maker_coin_htlc_key_pair {
-            Some(k) => Some(k.public_slice().into()),
-            None => Some(self.ctx.secp256k1_key_pair().public_slice().into()),
-        };
-
-        let taker_coin_htlc_pubkey = match taker_coin_htlc_key_pair {
-            Some(k) => Some(k.public_slice().into()),
-            None => Some(self.ctx.secp256k1_key_pair().public_slice().into()),
-        };
+        // Taker generates swap UUID so it's safe for him to use it for privkey derivation
```

Comment:
How about adding `TakerSwap::unique_swap_data` (returns `secret_hash`) and `MakerSwap::unique_swap_data` (returns `uuid`) methods?
If we change the unique swap data (let's say from uuid to random bytes) some day, it will help us to avoid mistakes.